docs/source/tests.rst

```diff
@@ -2,11 +2,20 @@
 Testing tools
 =============
 
+The best code comes with unit tests. Sopel provides a :doc:`pytest plugin
+<tests/pytest>` with standard fixtures; :doc:`factories <tests/factories>` and
+:doc:`mocks <tests/mocks>` to support writing tests; and the
+:func:`~.tests.rawlist` function to help analyze the mocked-up bot's output.
+
```

Comment:
I approve of the brief narrative added, do we also want to mention here that we do some test generation from plugin examples?

Reply:
Hmmmmmmmmmm, perhaps not _yet_. We still have that whole thing with `vcrpy` compatibility hanging over us at the moment, so I don't _really_ want to encourage wider use of the `@example` decorator just now. 😅

(I also go back and forth on whether `@example` even deserves to be part of the plugin API, beyond the basic "Here's how you call the command, and here's some sample output." for help reasons. Overloading it to generate tests is cool, but not something I'm positive that we'll want to keep long-term.)

Reply:
Sorry to hack the conversation to say: I even proposed, a year ago or so, to have 2 separate decorators, one for auto-test, one for actual usage examples, so the test framework doesn't clash with the documentation and the plugin definition. 😁